src/VisualStudio/Core/Def/Implementation/TaskList/ExternalErrorDiagnosticUpdateSource.cs

```diff
@@ -576,14 +598,17 @@ private sealed class InProgressState
 
             #endregion
 
-            public InProgressState(ExternalErrorDiagnosticUpdateSource owner, Solution solution)
+            public InProgressState(ExternalErrorDiagnosticUpdateSource owner, Solution solution, CancellationToken cancellationToken)
             {
                 _owner = owner;
                 Solution = solution;
+                CancellationToken = cancellationToken;
             }
 
             public Solution Solution { get; }
 
+            public CancellationToken CancellationToken { get; }
```

Comment:
dont' love this (as per higher conversation).  is it necessary for correctness of state/capturing?